internal/statuspage/api.go

```diff
@@ -0,0 +1,109 @@
+package statuspage
+
+import (
+	"fmt"
+	"github.com/broadinstitute/revere/internal/configuration"
+
+	"github.com/broadinstitute/revere/internal/shared"
+	"github.com/go-resty/resty/v2"
+)
+
+// Component represents how Statuspage returns components in its API.
+// This is an exact superset of what Statuspage accepts as input for components,
+// which is in turn an exact superset of pkg.Component.
+type Component struct {
+	AutomationEmail    string `json:"automation_email"`
+	CreatedAt          string `json:"created_at"`
+	Description        string `json:"description"`
+	Group              bool   `json:"group"`
+	GroupID            string `json:"group_id"`
+	ID                 string `json:"id"`
+	Name               string `json:"name"`
+	OnlyShowIfDegraded bool   `json:"only_show_if_degraded"`
+	PageID             string `json:"page_id"`
+	Position           int    `json:"position"`
+	Showcase           bool   `json:"showcase"`
+	StartDate          string `json:"start_date"`
+	Status             string `json:"status"`
+	UpdatedAt          string `json:"updated_at"`
+}
+
+func ComponentConfigToApi(configComponent configuration.Component, apiComponent *Component) {
```

Comment:
My preference for this type of type conversion function would be to have the `config.Component` as a function parameter and return a new instance of `apiCompnent` as output. It's possible there's is some piece I'm missing here though.

Reply:
Ooh this is a good catch, I'm adding a comment here and changing the function name. This function is actually used to modify a possibly existing Component, because this is how we detect if a Component requires a PATCH request to the server: we compare the Component and the component after calling this method and see if they're different.